### PR TITLE
Add XDebug

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

This PR adds XDebug's configuration files to Appwrite to leverage the XDebug extension implemented in the PR here: https://github.com/appwrite/docker-base/pull/20

## Test Plan

Build with the new docker-base and test if the debugger works

## Checklist

- [X] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -56,6 +56,7 @@ COPY ./public /usr/src/code/public
 COPY ./bin /usr/local/bin
 COPY ./docs /usr/src/code/docs
 COPY ./src /usr/src/code/src
+COPY ./dev /usr/src/code/dev
```
Comment (abnegate) [inline Dockerfile:59]: Let's remove this from the Dockerfile but mount it in docker-compose.yml, so that it's only present in dev mode

Comment (PineappleIOnic) [reply Dockerfile:59]: The only problem is that we copy the file into the php config directories later in these steps, unlike YASD there doesn't appear to be an init_file config for xdebug's ini files.

https://github.com/appwrite/appwrite/blob/77caf64076c6f56bc18bd173a538cd6775cd2000/Dockerfile#L114C7-L114C7

Comment (abnegate) [reply Dockerfile:59]: Makes sense, we can still remove it for non-debug, just needs an extra instruction _after_ the `COPY`:

Let's do the same to remove the xdebug extension file altogether if not in debug. Even if it's not loaded, it's bad practice to have XDebug present in production.

```dockerfile
RUN if [ "$DEBUG" = "false" ]; then rm -rf /usr/src/code/dev; fi
RUN if [ "$DEBUG" = "false" ]; then rm -f /usr/local/lib/php/extensions/no-debug-non-zts-20220829/xdebug.so
```

**Review by abnegate — COMMENTED** [review]
We can now update docker-base to [0.6.0](https://github.com/appwrite/docker-base/releases/tag/0.6.0)

**Review by abnegate — APPROVED** [review]

**Pull request merged**